source/client-side-encryption/client-side-encryption.rst

```diff
@@ -591,6 +623,59 @@ the user. Refer:
 - `Managing mongocryptd`_
 - `Detecting csfle Availability`_
 
+EncryptedFieldConfigMap
+^^^^^^^^^^^^^^^^^^^^^^^
+
+``EncryptedFieldConfigMap`` maps a collection namespace to an ``EncryptedFieldConfig``.
+
+``EncryptedFieldConfigMap`` only applies to FLE 2.
+
+If a collection is present on both the ``EncryptedFieldConfigMap`` and ``schemaMap``, libmongocrypt_ will error on initialization. See :ref:`fle2-and-fle1-error`.
+
+If a collection is present on the ``EncryptedFieldConfigMap``, the behavior of ``CreateCollection()`` and ``Collection.Drop()`` is altered. See :ref:`fle2-createcollection-drop`.
```

Comment:
To clarify my understanding, this will affect the regular driver `createCollection()`/`drop()` methods, and users won’t need to be calling a different method for creating FLE 2 collections? How does this map to the `ClientEncryption.createEncryptedCollection()` method lined out in the scope (i.e. is the expectation that the driver `createCollection()` method will call that)?

Reply:
> To clarify my understanding, this will affect the regular driver createCollection()/drop() methods, and users won’t need to be calling a different method for creating FLE 2 collections?
Yes. The proposal is to use `createCollection()` / `drop()` methods.

> How does this map to the ClientEncryption.createEncryptedCollection() method lined out in the scope (i.e. is the expectation that the driver createCollection() method will call that)?

The proposal is not to add `ClientEncryption.createEncryptedCollection()` / `ClientEncryption.dropEncryptedCollection` as is scoped. There is no longer a requirement to automatically create data keys, so I do not think the new API is necessary.

Reply:
> The proposal is not to add `ClientEncryption.createEncryptedCollection()` / `ClientEncryption.dropEncryptedCollection()` as is scoped.

Thank you for clariying. One final question then: Am I understanding correctly that this means that users have to know which FLE2 collections they will be creating at `MongoClient` instantiation time? That seems fine for applications, but might not be quite what users expect when they use a tool like mongosh or Compass.

Reply:
> Am I understanding correctly that this means that users have to know which FLE2 collections they will be creating at MongoClient instantiation time?

As proposed, yes.

> That seems fine for applications, but might not be quite what users expect when they use a tool like mongosh or Compass.

Good point. Another option is to add an `EncryptedFieldConfig` option to the `createCollection()` helper directly. I briefly considered it, but thought a new option would not be necessary. Does adding an `EncryptedFieldConfig` option to `CreateCollection` seem OK for mongosh and Compass use cases?

Reply:
Yeah, I think that would make sense. For drivers that already pass through options as they are, those could inspect that field to understand whether they would need to perform the additional steps lined out here, correct?

Reply:
> For drivers that already pass through options as they are, those could inspect that field to understand whether they would need to perform the additional steps lined out here, correct?

Yes, that seems OK. If the `encryptedFieldConfig` option is present, remove it, and do the steps to create the state collections and index.

I will proceed with proposing a new `EncryptedFieldConfig` option in `createCollection()`.

Reply:
For drop as well?

Reply:
> For drop as well?

Will do.

Reply:
That sounds okay – I think you could also just reuse `encryptedFields`, i.e. the options that are actually part of the `create` command, since they will contain the same information as the EncryptedFieldConfigMap for the collection?

Reply:
> That sounds okay – I think you could also just reuse encryptedFields, i.e. the options that are actually part of the create command, since they will contain the same information as the EncryptedFieldConfigMap for the collection?

Good point. I have renamed `EncryptedFieldConfig` to `EncryptedFields`.